runtimes/common/src/asset_switch/hooks.rs

```diff
@@ -16,29 +16,36 @@
 
 // If you feel like getting in touch with us, you can do so at info@botlabs.org
 
+use frame_support::traits::fungible::Inspect;
 use pallet_asset_switch::traits::SwitchHooks;
-use runtime_common::{AccountId, Balance};
+use sp_std::marker::PhantomData;
 use xcm::{
 	v4::{Junction, Junctions, Location},
 	VersionedLocation,
 };
 
-use crate::{KiltToEKiltSwitchPallet, Runtime};
+const LOG_TARGET: &str = "runtime::asset-switch::RestrictTransfersToSameUser";
 
-const LOG_TARGET: &str = "runtime::peregrine::asset-switch::RestrictTransfersToSameUser";
+type AccountIdOf<R> = <R as frame_system::Config>::AccountId;
+type BalanceOf<R, I> = <<R as pallet_asset_switch::Config<I>>::LocalCurrency as Inspect<AccountIdOf<R>>>::Balance;
 
 /// Check requiring the beneficiary be a single `AccountId32` junction
 /// containing the same account ID as the account on this chain initiating the
 /// switch.
-pub struct RestrictswitchDestinationToSelf;
-
-impl SwitchHooks<Runtime, KiltToEKiltSwitchPallet> for RestrictswitchDestinationToSelf {
+pub struct RestrictSwitchDestinationToSelf<R, P>(PhantomData<(R, P)>);
```

Comment:
The types are already present in the trait definition, you can keep the struct free of any type generics.

Reply:
Okay, but how should I provide the runtime and the pallet instance to the trait?

Reply:
Ah, yeah. Was brain afk: [d4e40e8](https://github.com/KILTprotocol/kilt-node/pull/701/commits/d4e40e8e72fd83f90c1f67c6c2813735932ddbb8)